lib/internal/crypto/hash.js

```diff
@@ -140,7 +145,53 @@ Hmac.prototype.digest = function digest(outputEncoding) {
 Hmac.prototype._flush = Hash.prototype._flush;
 Hmac.prototype._transform = Hash.prototype._transform;
 
+function Mac(mac, key, options) {
+  if (!(this instanceof Mac))
```

Comment:
I wonder if we can start getting away from having to do this sort of thing and even make this an actual ES6 class?

Reply:
Ideally, yes, but that's a semver-major change for the other classes in here (Hash, Hmac, etc). I had tried that in the original crypto refactor and it ended up breaking a bunch of stuff. Granted, however, that was a couple of years ago. I'm much happier with this revised API and impl but we'd have to decide if it's ok for new classes to use ES6 format while the existing ones use the legacy form.

Reply:
Right, I meant changing it for this new class, not changing existing classes.